presto-cli/src/main/java/com/facebook/presto/cli/QueryRunner.java

```diff
@@ -91,79 +96,26 @@ public Query startQuery(String query)
 
     public StatementClient startInternalQuery(String query)
     {
-        return new StatementClient(httpClient, queryResultsCodec, session.get(), query);
+        return new StatementClient(httpClient, session.get(), query);
     }
 
     @Override
     public void close()
     {
-        httpClient.close();
+        httpClient.dispatcher().executorService().shutdown();
```

Comment:
I think `QueryRunner` and `OkHttpClient` should have different lifecycle. You can use the same http client for multiple query runners. As far as i understand, after you do `httpClient.newBuilder()` it just copies the reference of the parent connection pool. So, once you close the connection pool for child, it will close is for the main client as well.